types/passport-azure-ad/oidc-strategy.d.ts

```diff
@@ -18,7 +18,7 @@ export interface IOIDCStrategyOption extends IBaseStrategyOption {
 }
 
 export interface IOIDCStrategyOptionWithRequest extends IOIDCStrategyOption {
-    passReqToCallback: true;
+    passReqToCallback: boolean;
```

Comment:
You should not change this. The `true` type for the property is required to pick a different overload in the constructor:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6148c753d84131f8c8581fa8974443e39d4d7a38/types/passport-azure-ad/oidc-strategy.d.ts#L56-L60

If you pass an `IOIDCStrategyOptionWithRequest` (which specifies `passReqToCallback: true`), then it picks the constructor overload that takes a `VerifyOIDCFunctionWithReq`. If you pass a regular `IOIDCStrategyOption`, it picks the overload that takes a `VerifyOIDCFunction`.

If the problem is that you need to pass `false` for `passReqToCallback` then you may need to add a separate `IOIDCStrategyOptionWithoutRequest` interface like this:

```ts
export interface IOIDCStrategyOptionWithoutRequest extends IOIDCStrategyOption {
  passReqToCallback?: false;
}
```

And then replace the constructor overload on Line 60:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6148c753d84131f8c8581fa8974443e39d4d7a38/types/passport-azure-ad/oidc-strategy.d.ts#L60

With this:

```ts
    constructor(options: IOIDCStrategyOptionWithoutRequest, verify: VerifyOIDCFunction);
```

Reply:
Thank you for your detailed comment!  I updated the `pasReqToCallback` type locally to `boolean` and I'm indeed passing `false`.  Things seem to be working ok but I have not tried passing `true` since that is not my use case.  Presumably the first constructor is still being called. It is somewhat confusing to me to see that `pasReqToCallback` is being used as a tag though since I was expecting it to be a simple property.  I will try your approach on my project and post my results.

Reply:
```suggestion
    passReqToCallback: true;
```

This needs to stay `true` for the correct overload selection to work.